apps/vscode/src/main.ts

```diff
@@ -338,34 +337,41 @@ async function registerWorkspaceFileWatcher(
   context: ExtensionContext,
   workspacePath: string
 ) {
-  if (workspaceFileWatcher) {
-    workspaceFileWatcher.dispose();
-  }
-
   const { nxWorkspace } = await import('@nx-console/shared/workspace');
 
   const { workspaceLayout } = await nxWorkspace(workspacePath, outputLogger);
   const workspacePackageDirs = new Set<string>();
   workspacePackageDirs.add(workspaceLayout.appsDir);
   workspacePackageDirs.add(workspaceLayout.libsDir);
   workspacePackageDirs.add('packages');
+
+  // build the base directory glob - will look something like this: `{appsDir,libsDir,packages}`
+  const basePackagesPath = Array.from(workspacePackageDirs).reduce(
+    (prev, current, index, array) => {
+      prev += `${current}`;
+      if (index === array.length - 1) {
+        prev += '}';
+      } else {
+        prev += ',';
+      }
+      return prev;
+    },
+    '{'
+  );
+
   context.subscriptions.push(
     watchFile(
-      new RelativePattern(workspacePath, '{workspace,angular,nx}.json'),
-      () => {
-        commands.executeCommand(REFRESH_WORKSPACE);
-      }
-    ),
-    ...Array.from(workspacePackageDirs).map((dir) => {
-      return watchFile(
-        new RelativePattern(
-          join(workspacePath, dir),
-          `**/{project,package}.json`
-        ),
-        () => {
-          commands.executeCommand(REFRESH_WORKSPACE);
-        }
-      );
+      new RelativePattern(workspacePath, '{workspace,angular,nx}.json')
+    )(() => {
+      commands.executeCommand(REFRESH_WORKSPACE);
+    }),
+    watchFile(
+      new RelativePattern(
+        workspacePath,
+        `${basePackagesPath}/**/{project,package}.json`
+      )
+    )(() => {
+      commands.executeCommand(REFRESH_WORKSPACE);
```

Comment:
Previously we created at max 3 file watches for each directory, but now it's always going to be one based on the glob.